src/Adapter/MSTest.CoreAdapter/Execution/TestExecutionRecorderWrapper.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution
+{
+    using System;
+    using System.Diagnostics;
+    using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Helpers;
+    using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+
+    /// <summary>
+    /// Wraps calls to ITestExecutionRecorder using types that can be serialized across AppDomains.
+    /// </summary>
+    internal class TestExecutionRecorderWrapper : MarshalByRefObject
```

Comment:
Unsure about the purpose of this class. Why couldn't we leverage the existing ITestExecutionRecorder  itself?

Reply:
@jayaranigarg the existing recorder can't be marshaled across app domains (notice this extends from MarshalByRefObject).